### PR TITLE
Two nitpicks

### PR DESCRIPTION
**Pull request by keszybz**

### CODE REVIEW

File: src/basic/unit-name.c [hunk 1]
```diff
@@ -246,11 +246,7 @@ int unit_name_build(const char *prefix, const char *instance, const char *suffix
         assert(prefix);
         assert(suffix);
         assert(ret);
-
-        if (isempty(suffix))
-                return -EINVAL;
-        if (suffix[0] != '.')
-                return -EINVAL;
+        assert(suffix[0] == '.');
```
Comment (filbranden) [inline src/basic/unit-name.c]: Asserts get compiled out if built with `-DNDEBUG` (see [docs](https://www.gnu.org/software/libc/manual/html_node/Consistency-Checking.html) and [header file](https://sourceware.org/git/?p=glibc.git;a=blob;f=assert/assert.h;h=5468c16c2c4e354e86454386c39a049daa130523;hb=23158b08a0908f381459f273a984c6fd328363cb#l43).)

Building with `-DNDEBUG` is a common setting, often pushed through CFLAGS in some build environments.

I agree with what @poettering said in the other thread: "place assert()s to catch programming errors while catching invalid data errors with runtime errors. I'd claim that a parameter being NULL where it shouldn't is really a programming error, while looking into a string could be an input data error, and hence should be handled more defensively..."

**Comment by poettering** [comment]
I agree with dropping the redundant check, but I still think we should invalid contents of the string as runtime error and not necessarily as fatal issue.

I mean, we have a tool like "systemd-escape" which can be used to put together unit names. Currently it doesn't expose this specific function, but who knows it might easily learn that, and I think it's much nicer to catch errors like this defensively...

**Comment by keszybz** [comment]
IMHO it's a case of "programming for the future", i.e. adding functionality which might be used potentially in the future, but has no use right now. All users right now are guaranteed to pass a dot-prefixed string. I imagine that if systemd-path starts to make use of this function, it'll ask for "service" or ".service" from the user, and turn that into ".service", and pass this into `unit_name_build`, without ever making use of the check in `unit_name_build`.

That said, I don't think this is important enough, so if you think y'all think it should stay a runtime check...

**Comment by keszybz** [comment]
Updated with the assert change reverted.

**Comment by poettering** [comment]
lgtm

**Pull request merged**